test/integ/cli/test_cli_incremental_download.py

```diff
@@ -0,0 +1,501 @@
+    downloaded_files = list(data_files)
+
+    verified_files = []
+
+    for downloaded_file in downloaded_files:
+        filename = downloaded_file.name
+        if filename in expected_files:
+            expected_marker = expected_files[filename]
+            content = downloaded_file.read_text()
+            # Verify file contains the original input
+            if "1. Input to CreateJob from data_dir" in content and expected_marker in content:
+                verified_files.append(filename)
+            else:
+                print(f"[dep_data_flow] WARNING: {filename} missing expected content")
+
+    # Verify all 8 expected files were created and have correct content
+    assert len(verified_files) == 8, (
+        f"Expected exactly 8 verified output files, found {len(verified_files)}: {verified_files}"
+    )
+
+    # Verify we have the complete set of expected files
+    missing_files = set(expected_files.keys()) - set(verified_files)
+    assert not missing_files, f"Missing expected files: {missing_files}"
+
+    print(
+        f"[dep_data_flow] Successfully verified all {len(verified_files)} expected output files with correct content"
+    )
+
+
+@pytest.mark.integ
+@pytest.mark.timeout(900)  # 15 minutes timeout
+def test_incremental_download_dependency_chain(incremental_download_test, tmp_path):
+    """Test incremental download with dep_chain template."""
+
+    unique_output_dir = f"{tmp_path}/dep_chain_output"
+
+    job_id = submit_dep_chain_job(
+        farm_id=incremental_download_test.farm_id,
+        queue_id=incremental_download_test.queue_id,
+        output_dir=unique_output_dir,
+    )
+
+    # Run incremental download in a loop until job completes
+    job_complete = False
+    force_bootstrap = True
+    while not job_complete:
+        job = incremental_download_test.deadline_client.get_job(
+            farmId=incremental_download_test.farm_id,
+            queueId=incremental_download_test.queue_id,
+            jobId=job_id,
+        )
+
+        task_run_status = job.get("taskRunStatus")
+
+        # Check if job failed
+        if task_run_status == "FAILED":
+            assert False, f"Job {job_id} failed with status: {task_run_status}"
+
+        incremental_download_test.run_incremental_download_without_storage_profiles(
+            str(tmp_path),
+            force_bootstrap=force_bootstrap,
+            lookback_window=2,
+            test_name="dep_chain",
+        )
+        force_bootstrap = False
+
+        job_complete = task_run_status in ["SUCCEEDED", "FAILED", "CANCELED"]
+        if not job_complete:
+            time.sleep(5)
+
+    # Run final incremental download to ensure all files are captured
+    time.sleep(5)
+    final_result = incremental_download_test.run_incremental_download_without_storage_profiles(
+        str(tmp_path), test_name="dep_chain"
+    )
+    assert final_result.returncode == 0, f"Final incremental download failed: {final_result.stderr}"
```

Comment:
nit: It looks like we have use this pattern in multiple tests. Would it be worth pulling it out into a separate function?